src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surName',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  sortOrder: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      studentsCopy.sort((firstStudent: Student, secondStudent: Student) => (
+        firstStudent.name.localeCompare(secondStudent.name)
+      ));
+      break;
+
+    case SortType.Surname:
+      studentsCopy.sort((firstStudent: Student, secondStudent: Student) => (
+        firstStudent.surname.localeCompare(secondStudent.surname)
+      ));
+      break;
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      studentsCopy.sort((firstStudent: Student, secondStudent: Student) => (
        firstStudent[sortBy].localeCompare(secondStudent[sortBy])
      ));
      break;
```

checklist p.3 [CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.